src/app/util/ClientMonitoringRegistrationTable.cpp

```diff
@@ -72,4 +96,11 @@ void ClientMonitoringRegistrationTable::setFabricIndex(FabricIndex fabric)
     mRegisteredClient.fabricIndex = fabric;
 }
 
+CHIP_ERROR ClientMonitoringRegistrationTable::IsRegisteredClientValid()
+{
+    return (mRegisteredClient.clientNodeId != 0 && mRegisteredClient.ICid != 0 && mRegisteredClient.fabricIndex != 0)
```

Comment:
A class function would be nice like

```
mRegisteredClient.IsValid()
```

Reply:
That attribute's type is a zap generated struct which doesn't have that function in the generation so can't really add this.

Reply:
You could subclass it and store the subclass and add the function on that.

Reply:
done